iothub_client/tests/common_longhaul/iothub_client_statistics.c

```diff
@@ -1468,15 +1471,16 @@ int iothub_client_statistics_get_device_twin_reported_summary(IOTHUB_CLIENT_STAT
                 summary->updates_sent--;
                 LogInfo("Removing twin reported update id (%d) because of [(%d) Request Timeout] network error", (int)device_twin_info->update_id, device_twin_info->send_status_code);
             }
-            else
+            else if (singlylinkedlist_find(stats->connection_status_history, compare_message_time_to_connection_time, &device_twin_info->time_queued))
             {
                 // check to see if the device was disconnected during this twin update
                 // we will miss the update because we reconnected to hub
-                if (singlylinkedlist_find(stats->connection_status_history, compare_message_time_to_connection_time, &device_twin_info->time_queued))
-                {
-                    summary->updates_sent--;
-                    LogInfo("Removing twin reported update id (%d) because of network error", (int)device_twin_info->update_id);
-                }
+                summary->updates_sent--;
+                LogInfo("Removing twin reported update id (%d) because of network error", (int)device_twin_info->update_id);
+            }
+            else
+            {
+                LogError("Error found in id (%d).", (int)device_twin_info->update_id);
```

Comment:
In this code path - and in general - it looks like the `result = 0` is being unconditionally set at end of function.  Shouldn't we return this error case(s) to caller?

Reply:
not really, there is parameter validation at the start. The error log is just there to help people know which iteration failed. Right now you have to search for it when it dumps out the result